dashboard/app/controllers/api/v1/pd/workshop_survey_report_controller.rb

```diff
@@ -1,91 +1,20 @@
 require_relative "#{Rails.root}/../pegasus/forms/pd_workshop_survey"
+require 'controllers/api/v1/pd/workshop_score_summarizer'
 
 class Api::V1::Pd::WorkshopSurveyReportController < Api::V1::Pd::ReportControllerBase
+  include WorkshopScoreSummarizer
   load_and_authorize_resource :workshop, class: 'Pd::Workshop'
 
-  FACILITATOR_EFFECTIVENESS_QUESTIONS = [
-      :how_much_learned_s,
-      :how_motivating_s,
-      :how_clearly_presented_s,
-      :how_interesting_s,
-      :how_often_given_feedback_s,
-      :help_quality_s,
-      :how_comfortable_asking_questions_s,
-      :how_often_taught_new_things_s
-  ]
-
-  TEACHER_ENGAGEMENT_QUESTIONS = [
-      :how_much_participated_s,
-      :how_often_talk_about_ideas_outside_s,
-      :how_often_lost_track_of_time_s,
-      :how_excited_before_s,
-      :overall_how_interested_s
-  ]
-
-  OVERALL_SUCCESS_QUESTIONS = [
-      :more_prepared_than_before_s,
-      :know_where_to_go_for_help_s,
-      :suitable_for_my_experience_s,
-      :would_recommend_s,
-      :part_of_community_s
-  ]
-
-  # There's some questions where a high number index is a bad thing (how often did you lose track of time?)
-  INVERTED_RESPONSE_QUESTIONS = [
-      :how_often_lost_track_of_time_s
-  ]
-
   # GET /api/v1/pd/workshops/:id/aggregate_workshop_score
   def workshop_survey_report
     survey_report = Hash.new
 
     survey_report[:this_workshop] = get_score_for_workshops([@workshop])
-    survey_report[:all_my_workshops] = get_score_for_workshops(Pd::Workshop.where(organizer_id: @workshop.organizer_id, course: @workshop.course))
-
-    render json: survey_report
-  end
-
-  private
-
-  def get_score_for_workshops(workshops)
-    report_column = Hash.new(0)
-    response_count = 0
+    survey_report[:all_my_workshops_for_course] = get_score_for_workshops(Pd::Workshop.where(organizer_id: @workshop.organizer_id, course: @workshop.course))
 
-    workshops.flat_map(&:enrollments).each do |enrollment|
-      response_object = PEGASUS_DB[:forms].where(source_id: enrollment.id, kind: 'PdWorkshopSurvey').first
+    aggregate_for_all_workshops = JSON.parse(AWS::S3.download_from_bucket('pd-workshop-surveys', 'aggregate-workshop-scores'))
```

Comment:
Since this S3 bucket will only be updated periodically (~once a day), we don't need to read it every time the controller action is executed. As a later optimization we should look into caching it in memory for some time.